demos/sev-demo/README.md

```diff
@@ -0,0 +1,136 @@
+
+## Rootfs
+ 
+Create the rootfs as you normally would for CCv0, but be sure to specify set `AA_KBC` to `offline_sev_kbc` so that the AA is built with the correct key broker client. 
+
+```
+sudo make USE_DOCKER=yes DISTRO=ubuntu AGENT_INIT=yes DOCKER_RUNTIME=runc AA_KBC=offline_sev_kbc SKOPEO_UMOCI=yes rootfs
+```
+
+You will need to add the agent configuration file to the initrd. Again, this is a typical CCv0 requirement, but you will need to specify parameters for you KBC. Specifically be sure to add this parameter.
+```
+aa_kbc_params = "offline_sev_kbc::null"
+```
+
+The `offline_sev_kbc` uses the `efi_secret` kernel module to expose injected secrets to the filesystem. You'll need to add the kernel module to the initrd. You can do this by setting the `KERNEL_MODULES_DIR` when you build your rootfs or you can copy the kernel module directly into the rootfs that is created. Either way, your rootfs should look something like this. 
+```
+$ tree ubuntu_rootfs/lib/modules
+lib/modules
+└── 5.15.0-rc5+
+    ├── kernel
+    │   └── drivers
+    │       └── virt
+    │           └── coco
+    │               └── efi_secret
+    │                   └── efi_secret.ko
+    ├── modules.alias
+    ├── modules.alias.bin
+    ├── modules.builtin
+    ├── modules.builtin.alias.bin
+    ├── modules.builtin.bin
+    ├── modules.builtin.modinfo
+    ├── modules.dep
+    ├── modules.dep.bin
+    ├── modules.devname
+    ├── modules.order
+    ├── modules.softdep
+    ├── modules.symbols
+    └── modules.symbols.bin
+```
+
+Make sure that the module matches your kernel version. Speaking of which, `efi_secret` also requires some minor kernel patches. These patches and the module are not currently upstream. You will need to find Dov Murik's EFI secret patches on the kernel mailing list and build a kernel with them. This should also give you the correct version of the `efi_secret` module.  
+
+
+## Kata Runtime
+
+Since SEV and SEV-ES use pre-attestation driven from outside of the guest, some changes are required for the Kata runtime. These changes are not currently upstream. You will need to build your Kata runtime with [this PR](https://github.com/kata-containers/kata-containers/pull/3025). This will work best with QEMU 6.2.0rc0. Newer versions of QEMU may require you to set `kernel-hashes=on` via the Kata runtime.
```

Comment:
Terminology: This is an awkward one. What do you mean by "Kata runtime"? I'm being awkward here but with good reason. The Kata Runtime for Kata 2.x is now known as the "Kata shim" (the `containerd-shim-kata-v2` binary). But Kata 2.x still provides a binary called `kata-runtime`, which is now just a utility program. It might be safer to refer to the "Kata Containers shim" here.

ftr, I'm planning to fill in the vast acreage of space in https://github.com/kata-containers/kata-containers/blob/main/Glossary.md to cover topics like this.

> Newer versions of QEMU may require you to set `kernel-hashes=on` via the Kata runtime.

This seems to be a QEMU option, but there is no way to set additional hypervisor options like this in the Kata configuration. It either requires the Kata runtime code be updated to add this, or you can use the interposing trick where you change the Kata config to specify the hypervisor as a shell script that exec's the hypervisor after first adding in additional options like this.

Reply:
Yeah I will switch to shim.

The extra QEMU flag is something that should be set by the runtime. We need to update our PR to do this. The QEMU change was merged in the middle of everything, hence to use the demo you need a very particular version of QEMU or you need to hackily add on the parameter.